packages/runner-ct/src/app/SpecList/useFuzzySort.ts

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react'
+import fuzzysort from 'fuzzysort'
+
+interface UseFuzzySort<Input, Result> {
+  search: string
+  items: Input[]
+  options: Fuzzysort.KeyOptions
+  transformResult: (spec: Input, indexes: number[]) => Result
+}
+
+export const useFuzzySort = <T, R>({ search, items, transformResult, options }: UseFuzzySort<T, R>) => {
+  const [lastSearchInput, setLastSearchInput] = React.useState<string>(search)
+  const [cachedFuzzySort, setCachedFuzzySort] = React.useState<R[]>([])
+  const [searchPromise, setSearchPromise] = React.useState<Fuzzysort.CancelablePromise<Fuzzysort.KeyResults<T>>>()
+
+  React.useLayoutEffect(() => {
```

Comment:
This does not need to be in a `useLayoutEffect`, and actually we specifically don't want it to be. `useLayoutEffect` will force a second render (of course assuming that it sets some state) in the same tick, so that you only end up drawing the second render.

The whole reason to put this logic into a `useEffect` in the first place is to avoid blocking rendering _something_ (in this case, the search box input) on the filtering

Reply:
🤯